plugins/Marketplace/API.php

```diff
@@ -177,15 +177,25 @@ public function deleteLicenseKey()
      * @unsanitized
      * @internal
      */
-    public function requestTrial(string $pluginName, string $pluginDisplayName = ''): bool
+    public function requestTrial(string $pluginName): bool
     {
         Piwik::checkUserIsNotAnonymous();
 
         if (Piwik::hasUserSuperUserAccess()) {
             throw new Exception('Cannot request trial as a super user');
         }
 
-        $this->pluginTrialService->request($pluginName, $pluginDisplayName);
+        if (!$this->pluginManager->isValidPluginName($pluginName)) {
+            throw new Exception('Invalid plugin name given ' . $pluginName);
+        }
+
+        $pluginInfo = $this->marketplaceClient->getPluginInfo($pluginName);
+
+        if (empty($pluginInfo['name'])) {
+            throw new Exception('Unable to find plugin with given name: ' . $pluginName);
+        }
+
+        $this->pluginTrialService->request($pluginInfo['name'], $pluginInfo['displayName'] ?? $pluginInfo['name']);
```

Comment:
Would it make sense to use the Elvis operator here? To catch not just null values, but also anything falsey?

```suggestion
        $this->pluginTrialService->request($pluginInfo['name'], $pluginInfo['displayName'] ?: $pluginInfo['name']);
```